### ISSUE TITLE
Error executing 'postInstallation'

### ISSUE THREAD

**Issue by grayloon** [bitnami/bitnami-docker-drupal#22]
```
# mkdir mydrupalapp
# cd mydrupalapp
# curl -L "https://raw.githubusercontent.com/bitnami/bitnami-docker-wordpress/master/docker-compose.yml" > docker-compose.yml
# mkdir {mariadb,drupal,apache,php}
```

Edit docker-compose.yml with the persistent volume paths:
- './mariadb:/bitnami/mariadb'
- './drupal:/bitnami/drupal'
- './apache:/bitnami/apache'
- './php:/bitnami/php'

```
# docker-compose up
```

```
application_1  | drupal  INFO  Updating Drupal database ...
application_1  | Error executing 'postInstallation': EACCES: permission denied, open '/opt/bitnami/drupal/sites/default/settings.php'
```

MacOS X 10.11.6
Docker version 1.12.0, build 8eab29e
docker-compose version 1.8.0, build f3628c7
docker-machine version 0.8.0, build b85aac1

docker info
Containers: 7
 Running: 0
 Paused: 0
 Stopped: 7
Images: 7
Server Version: 1.12.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 73
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.15-moby
Operating System: Alpine Linux v3.4
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.954 GiB
Name: moby
ID: PI3T:NGOO:ICYA:ZE6A:C43S:KBOB:3XHH:I5XD:MLTH:FZIP:BFLM:HUWL
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 18
 Goroutines: 29
 System Time: 2016-09-14T19:32:11.261381109Z
 EventsListeners: 1
No Proxy: *.local, 169.254/16
Username: cdoggyd
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8

**Comment by prydonius** [comment]
Thanks @grayloon, I tested on Debian (worked fine) and Docker for Mac (I was able to reproduce the issue). We've run into permissions issues volumed mounted over VirtualBox shared folders, or NFS, but this is the first time I'm seeing issues with Docker for Mac.

I'll investigate to see if there's anything we can do.

/cc @sameersbn

**Comment by sameersbn** [comment]
I believe this is an issue with Docker for Mac. I found that if a host volume is mounted in a container, then you cannot change the ownership of a file in the volume that has read-only mode.

fe.

```bash
$ docker run -it --rm $PWD:/data ubuntu:14.04 bash
$ touch sample && chmod 444 sample && ls -la
total 5
drwxr-xr-x  3 root root  102 Sep 15 12:28 .
drwxr-xr-x 35 root root 4096 Sep 15 12:27 ..
-r--r--r--  1 root root    0 Sep 15 12:28 sample
$ chown daemon:daemon sample
chown: changing ownership of 'sample': Permission denied
```

The image encounters this issue while trying to set the file ownerships and permissions during the post-installation setup. A possible workaround is to set the mode of the files and directories to `read-write` mode before changing the ownerships (after which the mode and be changed back to read-only).

**Comment by sameersbn** [comment]
FYI, reported the issue @ https://github.com/docker/for-mac/issues/633

**Comment by sameersbn** [comment]
@grayloon The referenced issue in docker for mac has been resolved and I am able to confirm it resolves the issue in Docker for Mac (stable) `1.12.5`.

Closing this issue. Please reopen if the issue persists.
Thank you for choosing Bitnami application images.
